lib/components/notice/notice.less

```diff
@@ -17,15 +17,9 @@
         --_no-code-bc: ~"var(--@{colorName}-300)";
         --_no-code-bg: ~"var(--@{colorName}-200)";
 
-        .dark-mode({
-            --_no-bc: ~"@{darkMode}";
-        });
-
-        .highcontrast-dark-mode({
-                --_no-bc: ~"@{highContrast}"
+        .highcontrast-mode({
+            --_no-bc: ~"var(--@{colorName}-400)";
```

Comment:
@ttaylor-stack since all high contrast mode borders should be the `400` color stop for their given color, we can remove the need for the extra `@highContrast` param and just concat the color name to `-400`.